setup.py

```diff
@@ -65,7 +65,6 @@ def rglob(path, glob):
         "numpy",
         "networkx",
         "omlt==0.3", # fix the version for now as package evolves
-        "tensorflow",
```

Comment:
@lbianchi-lbl: I am all for culling required dependencies, but as we do that, should we start an "optional" target with something like:
```
extras_require = {
    'optional': [
        'tensorflow',  # idaes.surrogates.keras_surrogate
    ]
}
```

Reply:
Yeah, I think it's a good idea. I thought we could deal with the optional dependencies all at once, but this looks like a good place to start.

Reply:
@carldlaird @jsiirola I can do this as part of #633. I propose we get this merged in as soon as all the tests pass.